src/Entity/Transformer.php

```diff
@@ -155,40 +204,51 @@ public function loadBlock(array $content, PageInterface $page): PageBlockInterfa
 
         $block->setPage($page);
 
-        if (isset($content['id'])) {
-            $block->setId($content['id']);
-        }
+        $blockClass = $this->blockManager->getClass();
+
+        // NEXT_MAJOR: Remove null support
+        if (null !== $this->serializer) {
+            $this->serializer->denormalize($content, $blockClass, null, [
+                DateTimeNormalizer::FORMAT_KEY => 'U',
+                AbstractNormalizer::OBJECT_TO_POPULATE => $block,
+                AbstractNormalizer::CALLBACKS => $this->getDenormalizeCallbacks(),
+            ]);
+        } else {
```

Comment:
and add a deprecation
```suggestion
        } else {
            if (!method_exists('removeChild', $page)) {
                trigger_error('Not implementing a `PageInterface::removeChild` method is deprecated since 4.x and will throw an error in 5.0.', \E_USER_DEPRECATED));
            }
            if (!method_exists('removeBlock', $page)) {
                trigger_error('...', \E_USER_DEPRECATED);
            }
```